tensorboard/components/experimental/plugin_util/plugin-host-ipc.ts

```diff
@@ -42,10 +51,33 @@ namespace tb_plugin.host {
     port.start();
 
     for (const [type, callback] of listeners) {
-      portIPC.listen(type, callback);
+      const callbackWithContext = wrapCallbackWithContext(callback, portIPC);
+      portIPC.listen(type, callbackWithContext);
     }
   }
 
+  /**
+   * Provides context data from the IPC to the callback.
+   */
+  function wrapCallbackWithContext(
+    callback: PluginHostCallback,
+    ipc: lib.DO_NOT_USE_INTERNAL.IPC
+  ): lib.DO_NOT_USE_INTERNAL.MessageCallback {
+    return (payload) => {
+      const context: PluginHostMessageContext = {pluginName: null};
+
+      // 'name' is a keyword set in the iframe URL by tf-tensorboard's
+      // `_renderPluginIframe`.
+      const frame = ipcToFrame.get(ipc);
+      if (frame) {
+        const pluginName = new URL(frame.src).searchParams.get('name');
```

Comment:
I believe this is modifiable by the plugin. I think there should be some global map of instance of an iframe to plugin metadata. https://github.com/tensorflow/tensorboard/blob/893775b8e0d4236ad6ee2082018aae672fcdbe48/tensorboard/components/tf_tensorboard/tf-tensorboard.html#L864-L871

We can of course do this later as it would be a larger surgery.

Reply:
Done in this PR